timelock-impl/src/main/java/com/palantir/atlasdb/timelock/auth/config/NamespacePrivileges.java

```diff
@@ -26,15 +26,21 @@
 @Value.Immutable
 public abstract class NamespacePrivileges implements Privileges {
     abstract Set<TimelockNamespace> namespaces();
+    abstract Set<String> namespaceSuffixes();
 
-    static Privileges of(Set<TimelockNamespace> namespaces) {
+    static Privileges of(Set<TimelockNamespace> namespaces, Set<String> namespaceSuffixes) {
         return ImmutableNamespacePrivileges.builder()
                 .addAllNamespaces(namespaces)
+                .namespaceSuffixes(namespaceSuffixes)
                 .build();
     }
 
     @Override
     public boolean hasPrivilege(TimelockNamespace namespace) {
-        return namespaces().contains(namespace);
+        return namespaces().contains(namespace) || hasMatchingSuffix(namespace);
+    }
+
+    private boolean hasMatchingSuffix(TimelockNamespace namespace) {
+        return namespaceSuffixes().stream().anyMatch(suffix -> namespace.value().endsWith(suffix));
```

Comment:
should it be an error for the suffix to be the namespace i.e. if suffix is "_alta", someone coming through with "_alta" as a namespace?